algorithms/src/Kokkos_Sort.hpp

```diff
@@ -624,6 +629,19 @@ std::enable_if_t<Kokkos::is_execution_space<ExecutionSpace>::value> sort(
   bin_sort.sort(exec, view);
 }
 
+#if defined(KOKKOS_ENABLE_CUDA)
+template <class ViewType>
+void sort(Cuda const& space, ViewType const& view) {
+  static_assert(ViewType::rank == 1, "CUDA sort only supports 1D Views.");
```

Comment:
It looks like the `View` needs to be contiguous in memory, too, right? 
```suggestion
  static_assert(ViewType::rank == 1 && !std::is_same<typename ViewType::array_layout, LayoutStride>::value, "CUDA sort only supports contiguous 1D Views.");
```

Reply:
the static assert has advantage of producing a readable message, the sfinea does not

Reply:
I agree. I was thinking in terms of a "fallback" - if "CUDA" requirements are not met then "original" implementation could be used.

Reply:
I would use a `static_assert` for the restrictions that apply to all overloads and `std::enable_if`/SFINAE if there is a fallback. We assume for all `Kokkos::sort` overloads that we have a 1D-View but only this overload is restricted to contiguous memory, right?

An alternative to restricting this to contiguous memory would be to use `RandomAccessIterator`.

Reply:
I used `RandomAccessIterator`.